action/protocol/staking/contractstaking/cache.go

```diff
@@ -156,16 +218,22 @@ func (s *contractStakingCache) mustGetBucketInfo(id uint64) *bucketInfo {
 	return bt
 }
 
-func (s *contractStakingCache) getBucket(id uint64) (*Bucket, error) {
+func (s *contractStakingCache) mustGetBucket(id uint64) *Bucket {
+	bi := s.mustGetBucketInfo(id)
+	bt := s.mustGetBucketType(bi.TypeIndex)
+	return assembleBucket(id, bi, bt)
+}
+
+func (s *contractStakingCache) getBucket(id uint64) (*Bucket, bool) {
 	bi, ok := s.getBucketInfo(id)
 	if !ok {
-		return nil, errors.Wrapf(ErrBucketNotExist, "id %d", id)
+		return nil, false
 	}
 	bt := s.mustGetBucketType(bi.TypeIndex)
-	return assembleBucket(id, bi, bt)
+	return assembleBucket(id, bi, bt), true
 }
 
-func (s *contractStakingCache) getTotalBucketTypeCount() uint64 {
+func (s *contractStakingCache) GetTotalBucketTypeCount() uint64 {
```

Comment:
rename to `BucketTypeCount`, and move to public function group

Reply:
fixed